File: api/Ez.Hress.Hardhead/Entities/YearEntity.cs
using Ez.Hress.Shared.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ez.Hress.Hardhead.Entities;

public class YearEntity : EntityBase<int>
{
    public int GuestCount { get; set; }
    public int? PhotoID { private get; set; }
    public HrefEntity? Photo
    {
        get
        {
            if (PhotoID.HasValue)
                return new HrefEntity()
                {
                    Href = string.Format("/api/images/{0}/content", PhotoID.Value)
                };

            return null;
        }
    }

    public UserBasicEntity? Hardhead { get; set; }
}

File: src/types/yearEntity.ts
import { EntityBase } from "./entityBase";
import { HrefEntity } from "./hrefEntity";
import { UserBasicEntity } from "./userBasicEntity";

export interface YearEntity extends EntityBase<number> {
  guestCount: number;
  description: string;
  photoId?: number;
  photo?: HrefEntity;
  hardhead: UserBasicEntity;
}
